### ISSUE TITLE
Unable to connect Sparrow to electrs via local connection or Tor

### ISSUE THREAD

**Issue by harkousb** [raspibolt/raspibolt#1273]
### Describe the issue
Stuck trying to connect Sparrow desktop wallet to electrs (https://raspibolt.org/guide/bitcoin/desktop-wallet.html#connect-to-electrs).
bitcoind, electrs, and nginx are running without errors. Indexing completed.

error when trying to connect via local connection:
"Could not connect:

Retries exhausted"

error when trying to connect via Tor:
"Could not connect:

SOCKS: Host unreachable. Check if the proxy server is running."

I tried the troubleshooting in the guide, making sure my computer is not connected to any “guest" network, and deleting the existing certificates that are stored within the certs folder. But, still unable to connect.

### Location in guide
(https://raspibolt.org/guide/bitcoin/desktop-wallet.html#connect-to-electrs)

### Environment
 - Raspberry Pi 4 B
 - Operating system: Raspberry Pi OS 64bit
 - Version: Debian GNU/Linux 11 (bullseye)

**Comment by harkousb** [comment]
I set log_filters = "DEBUG" in electrs.conf, restarted the service. 

running "sudo journalctl -f -u electrs" while trying to connect:

May 01 16:52:51 raspi electrs[1770]: [2023-05-01T23:52:51.809Z ERROR electrs::server] 6: disconnecting due to failed to send response: "{\"id\":3,\"jsonrpc\":\"2.0\",\"result\":{\"height\":782590,\"hex\":\"0000c020fd4b4b7c34983310f187a14106292c288660df6dbd8e020000000000000000007396eb361ae715d6069048e77e752740930887cef866c21aa003c9be76e2c9a8f01a20643e020617413ea70a\"}}\n"
May 01 16:52:51 raspi electrs[1770]: [2023-05-01T23:52:51.809Z WARN  electrs::server] 6: failed to shutdown TCP connection Transport endpoint is not connected (os error 107)

running "sudo journalctl -f -u electrs | grep :server" while trying to connect:
May 01 16:49:55 raspi electrs[1770]: [2023-05-01T23:49:55.828Z ERROR electrs::server] 5: disconnecting due to failed to send response: "{\"id\":3,\"jsonrpc\":\"2.0\",\"result\":{\"height\":780590,\"hex\":\"00805320795ae2eb6a52f8cf757469b182d31eb850ed1c64153d01000000000000000000f9a40be591ef086d82ffe2740567ff2c2fe69df179cb185de37c558f0236145dabdb0e6481760617400620d6\"}}\n"
May 01 16:49:55 raspi electrs[1770]: [2023-05-01T23:49:55.828Z WARN  electrs::server] 5: failed to shutdown TCP connection Transport endpoint is not connected (os error 107)

**Comment by zynos** [comment]
And you checked that the elctrs port is enabled in ufw?
Although it should work with tor anyway..

**Comment by dutu** [comment]
Try connecting with Electrum. Electrum worked for me, but I had intermittent issues with Sparrow for some reasons.
That's why I decided to move to Fulcrum.

**Comment by zynos** [comment]
Strange, for me its working, i also tested it with the newest sparrow version again..

**Comment by harkousb** [comment]
@zynos yes I double checked this step:
$ sudo ufw allow 50002/tcp comment 'allow Electrum SSL'

@dutu I tried connecting with Electrum as well, same problem.

**Comment by zynos** [comment]
What is your version of BTC Core? Mine is 23.
You can also have a look here:
https://community.getumbrel.com/t/sparrow-wallet-desktop/6898/24

**Comment by harkousb** [comment]
@zynos v24.0.1, I tried switching to 23.1 but had the same issue, so I switched back to 24.0.1 again.

I tried running the [RaspiBolt System Overview](https://raspibolt.org/guide/bonus/raspberry-pi/system-overview.html) and it gets stuck at loading Electrum Server Data
![image](https://user-images.githubusercontent.com/39776417/236496885-45070081-0b51-4c0b-b5f2-144bce25c21a.png)

Running `sudo journalctl -f -u electrs | grep :server` with the log_filters = "DEBUG" in electrs.conf file while trying to connect using Sparrow shows this:
![image](https://user-images.githubusercontent.com/39776417/236498192-eeabc94a-f6d1-4555-8299-8c777af684f6.png)

On the Sparrow side:
![image](https://user-images.githubusercontent.com/39776417/236498394-489d12a0-33ee-42c7-a010-09df2764f7b0.png)

Any ideas what could be causing this? Seems to me like it's an issue with my electrs.
